cmd/livepeer_router/livepeer_router.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"flag"
+	"net/url"
+	"os"
+	"os/signal"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/server"
+)
+
+func defaultAddr(addr, defaultHost, defaultPort string) string {
+	if addr == "" {
+		return defaultHost + ":" + defaultPort
+	}
+	if addr[0] == ':' {
+		return defaultHost + addr
+	}
+	// not IPv6 safe
+	if !strings.Contains(addr, ":") {
+		return addr + ":" + defaultPort
+	}
+	return addr
+}
+
+func main() {
+	flag.Set("logtostderr", "true")
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	datadir := flag.String("datadir", "", "Data directory")
```

Comment:
It seems like the usage of this flag is to be able to use a pre-signed certificate by a CA rather than generate a self-signed one. The naming and description however makes it a little opaque that this is the intended usage. 

Perhaps we could update the description to something like "Directory for SSL certificate and key, will be auto-generated if no certificate is available". 

Alternatively splitting up the flags into `-certificate`, `-certKey` with paths to the files rather than the directory or something along those lines (this might also help prevent overwriting other certificates if the user can bring its own filenames).

Reply:
Actually the `-datadir` flag here serves the same purpose as in `livepeer`. The flag is used to define a directory where all data for the binary should be stored. In the case of `livepeer_router`, the directory is just used to store the self-signed cert that is generated at startup (as is the case for `livepeer` as well), but it could be used to store other data as well if needed. I tried to make the description slightly clearer for `livepeer_router` in 990363c as well as the description for `livepeer` in ec7a640 to match. Open to other suggestions.

Neither `livepeer` nor `livepeer_router` support user defined certs right now because a self-signed cert is always generated at startup via the [getCert()](https://github.com/livepeer/go-livepeer/blob/master/server/cert.go#L82) function. I did not intent to include that feature in this PR, but it'd be a nice feature to consider.